app/models/spree/product_decorator.rb

```diff
@@ -115,6 +115,13 @@
     end
   }
 
+  scope :stockable_by, lambda { |enterprise|
+    return where('1=0') unless enterprise.present?
```

Comment:
ActiveRecord in Rails 4 has `.none` that's equivalent to `where('1=0')`. Not yet, but soon...